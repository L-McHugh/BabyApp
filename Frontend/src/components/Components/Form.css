.formDiv {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 2vh;
  margin-bottom: 2vh;
}

.timeLabel {
  font-size: 1.5em;
  font-weight: bold;
  text-align: center;
}

.typeLabel {
  font-size: 1.5em;
  font-weight: bold;
  text-align: center;
  margin-left: 2vw;
  margin-right: 2vw;
}

input {
  border-radius: 5px;
  font-size: auto;
  padding: 0.5vw;
}
