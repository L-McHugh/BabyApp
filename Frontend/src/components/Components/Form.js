import React from "react";
import "./Form.css";

export default function Form(props) {
  return (
    <div>
      <div data-testid="form" className="exitButtonDiv">
        <button className="exitButton" onClick={props.visibility}>
          X
        </button>
      </div>

      <form className="formDiv" autoComplete="off">
        <label className="timeLabel">
          Start time:
          <input
            data-testid="formTime"
            type="datetime-local"
            name="start_time"
            required
            onChange={props.onChangeTime}
          />
        </label>

        <label className="typeLabel">
          {props.label}:
          <input
            data-testid="formDescription"
            type={props.type}
            name="description"
            required
            onChange={props.onChangeDescription}
            placeholder={props.placeholder}
          />
        </label>

        <button data-testid="formButton" onClick={props.onClick}>Submit</button>
      </form>
    </div>
  );
}
