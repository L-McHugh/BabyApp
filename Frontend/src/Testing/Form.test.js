import React from "react";
import "@testing-library/jest-dom";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Form from "../components/Components/Form";

describe("<Form/>", () => {
  const mockedProps = {
    label: "Description",
    type: "text",
    placeholder: "Enter description",
    onChangeTime: jest.fn(),
    onChangeDescription: jest.fn(),
    onClick: jest.fn(),
  };

  it("successfully renders the component, matches snapshot", () => {
    const { asFragment } = render(<Form {...mockedProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct type", () => {
    render(<Form {...mockedProps} />);
    expect(screen.getByTestId("formDescription")).toHaveAttribute(
      "type",
      "text"
    );
  });

  it("renders the correct placeholder", () => {
    render(<Form {...mockedProps} />);
    expect(screen.getByTestId("formDescription")).toHaveAttribute(
      "placeholder",
      "Enter description"
    );
  });

  it("renders the correct button", () => {
    render(<Form {...mockedProps} />);
    expect(screen.getByTestId("formButton")).toHaveTextContent("Submit");
  });

  it("calls the correct function when the button is clicked", () => {
    render(<Form {...mockedProps} />);
    userEvent.click(screen.getByTestId("formButton"));
    expect(mockedProps.onClick).toHaveBeenCalledTimes(1);
  });

  it("calls the correct function when the description is changed", () => {
    render(<Form {...mockedProps} />);
    userEvent.type(screen.getByTestId("formDescription"), "Test description");
    expect(mockedProps.onChangeDescription).toHaveBeenCalledTimes(16);
  });
});
