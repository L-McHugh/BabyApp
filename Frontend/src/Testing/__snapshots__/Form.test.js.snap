// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> successfully renders the component, matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="exitButtonDiv"
      data-testid="form"
    >
      <button
        class="exitButton"
      >
        X
      </button>
    </div>
    <form
      autocomplete="off"
      class="formDiv"
    >
      <label
        class="timeLabel"
      >
        Start time:
        <input
          data-testid="formTime"
          name="start_time"
          required=""
          type="datetime-local"
        />
      </label>
      <label
        class="typeLabel"
      >
        Description:
        <input
          data-testid="formDescription"
          name="description"
          placeholder="Enter description"
          required=""
          type="text"
        />
      </label>
      <button
        data-testid="formButton"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
